metatron/src/test/java/uk/gov/ida/eidas/metatron/apprule/rules/MetatronTests.java

```diff
@@ -0,0 +1,263 @@
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    public static RuleChain init() throws Exception {
+        testMetadataResource = new TestMetadataResource();
+        testMetadataServer =  new CreatesConfigDropwizardClientRule(testMetadataResource);
+        waitWhile(() -> testMetadataServer.getPort() == 0, "Waiting for client rule port ");
+
+        testMetadataResource.init(testMetadataServer.getPort());
+        metatronAppRule = new MetatronAppRule();
+
+        return RuleChain.outerRule(testMetadataServer).around(metatronAppRule);
+    }
+
+    private static void waitWhile(Supplier<Boolean> condition, String message) {
+        // This needs a time-out mechanism
+        System.out.print(message + " ");
+        while(condition.get()) {
+            System.out.println(".");
+        }
+        System.out.println("\nDone");
+    }
+
+    @Test
+    public void firstValidResolvesOk() {
+        String entityId = getEntityId(TestMetadataResource.VALID_ONE);
+        Response response = getClient().target(getUriString(entityId)).request().get();
+
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.OK);
+        assertThat(response.readEntity(String.class)).isEqualTo(entityId);
+    }
+
+    @Test
+    public void secondValidResolvesOk() {
+        String entityId = getEntityId(TestMetadataResource.VALID_TWO);
+        Response response = getClient().target(getUriString(entityId)).request().get();
+
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.OK);
+        assertThat(response.readEntity(String.class)).isEqualTo(entityId);
+    }
+
+    @Test
+    public void expiredReturns500() {
+        String entityId = getEntityId(TestMetadataResource.EXPIRED);
+        Response response = getClient().target(getUriString(entityId)).request().get();
+
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.INTERNAL_SERVER_ERROR);
+    }
+
+    @Test
+    public void untrustedReturns500() {
+        String entityId = getEntityId(TestMetadataResource.SIGNED_UNTRUSTED);
+        Response response = getClient().target(getUriString(entityId)).request().get();
+
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.INTERNAL_SERVER_ERROR);
+    }
+
+    @Test
+    public void unsignedReturns500() {
+        String entityId = getEntityId(TestMetadataResource.UNSIGNED);
+        Response response = getClient().target(getUriString(entityId)).request().get();
+
+        assertThat(response.getStatusInfo()).isEqualTo(Response.Status.INTERNAL_SERVER_ERROR);
+    }
+
+    private String getEntityId(String name) {
+        return "http://localhost:" + testMetadataServer.getPort() + "/application/" + name + "/Metadata";
+    }
+
+    private String getUriString(String entityId) {
+        return "http://" + UriBuilder.fromMethod(MetatronResource.class, "metadata").host("localhost").port(metatronAppRule.getLocalPort()).build(entityId).toString();
```

Comment:
This is interesting - looks like it URL encodes the entityId automatically. Might be an update to the client I merged today.

Reply:
I suspect it's not limited to the .fromMethod() method would probably work a treat.